src/components/_product/HeroPhysical/HeroPhysical.tsx

```diff
@@ -0,0 +1,217 @@
+    chosenVariant?.gallery!.forEach((el) => images.push({ type: 'image', data: el }));
+    return images;
+  }, [chosenVariant]);
+
+  const selectVariant = useCallback(
+    (v: SingleValue<{ value: string | undefined; label: string | undefined }>, name: string) => {
+      const filteredAttributes = { ...chosenAttributes };
+      filteredAttributes[name] = v?.value;
+
+      // find variant with all the same attributes
+      const filteredVariant = variants.find((el) => {
+        let all = true;
+        el.attributes!.every((attr) => {
+          if (filteredAttributes[attr.name.toLocaleLowerCase()] !== attr.value) all = false;
+
+          return all;
+        });
+        return all;
+      });
+
+      setChosenVariant(filteredVariant);
+      setChosenAttributes(filteredAttributes);
+      setSelectedImage(0);
+      setCount(filteredVariant!.countInStock > 0 ? 1 : 0);
+    },
+    [variants, chosenAttributes]
+  );
+
+  return (
+    <section className={styles['HeroPhysical']}>
+      <div className={styles['gallery']}>
+        {gallerySwitch(images[selectedImage]!.data)[images[selectedImage]!.type as 'image' | 'video']}
+        <div className={styles['gallery-grid']}>
+          {images.map((el, index) => {
+            if (index === selectedImage) return null;
+            return (
+              <button
+                onClick={() => setSelectedImage(index)}
+                key={index}
+              >
+                {gallerySwitch(el.data)[el.type as 'image' | 'video']}
+              </button>
+            );
+          })}
+        </div>
+        {/* TODO: add arrows */}
+      </div>
+      <div className={styles['info']}>
+        {/* <p>reviews</p> TODO: add reviews */}
+        <h1>{name}</h1>
+        <div className={styles.attributes}>
+          {attributes.map((el) => (
+            <p key={el.name}>
+              <span>Wybierz {el.name}</span>
+              <Select
+                defaultValue={{ value: chosenAttributes[el.name], label: chosenAttributes[el.name] }}
+                onChange={(e) => selectVariant(e, el.name)}
+                options={(() => [...el.value.map((v) => ({ value: v, label: v }))])()}
+              />
+            </p>
+          ))}
+        </div>
+        <div className={styles['flex']}>
+          <div className={styles['calculator']}>
+            <div>
+              <button
+                disabled={count <= 1}
+                onClick={() => {
+                  setCount(count - 1);
+                }}
+              >
+                <svg
+                  xmlns='http://www.w3.org/2000/svg'
+                  width='13'
+                  height='2'
+                  fill='none'
+                >
+                  <path
+                    stroke='#9A827A'
+                    d='M12.3.7H1'
+                    strokeLinecap='round'
+                    strokeLinejoin='round'
+                  />
+                </svg>
```

Comment:
I would rather declare `SVG` on `server` componenet and pass it to a `client` component as a props. 
Moreover: it should be optimized with [SVGOMG](https://svgomg.net/)

Reply:
I don’t think that in this case the time spent will correspond to the benefit gained.

P.S. Already optimized.